harness/tests/common/api/test_authentication.py

```diff
@@ -125,3 +127,30 @@ def test_auth_json_v1_upgrade() -> None:
         # Make sure we got exactly the master urls we expected.
         exp = {"http://firstmaster:80", "https://secondmaster:443", "http://thirdmaster:8080"}
         assert set(obj["masters"]) == exp, list(obj["masters"])
+
+
+def test_salt_and_hash() -> None:
+    assert authentication.salt_and_hash("") == ""
+    assert (
+        authentication.salt_and_hash("A05850A9-bcfe-4ea6-8363-7350ee604be7")
+        == "c7e7c0ffdcdb574914b3c36dc6deb37a37e5c36f25dabef323cf470041c122"
+        + "470b3beab157cb8c1624cab6dd9d8e27dd242a128e5899a5c8b7fb7a1b7c521778"
+    )
+
+
+@pytest.mark.parametrize(
+    "password, should_raise, error",
+    [
+        ("F7A90041-9f0d-4dfe-8cba-21555072ccdf", False, None),
+        ("pass", True, "password must have at least 8 characters"),
```

Comment:
I think it's marginally less implementation-dependent (and therefore less brittle) to have a short password that passes the other tests.